pkg/configv2/default.go

```diff
@@ -0,0 +1,92 @@
+package configv2
+
+import (
+	"runtime"
+	"time"
+
+	"github.com/bacalhau-project/bacalhau/pkg/configv2/types"
+)
+
+var Default = types.Bacalhau{
+	Repo: "~/.bacalhau",
+	Name: "",
+	Client: types.Client{
+		Address: "http://0.0.0.0:1234",
+	},
+	Server: types.Server{
+		Address: "0.0.0.0",
+		Port:    1234,
+		TLS:     types.TLS{},
+		Auth: types.AuthConfig{
+			Methods: map[string]types.AuthenticatorConfig{
+				"ClientKey": {
+					Type: "challenge",
+				},
+			},
+		},
+	},
+	Orchestrator: types.Orchestrator{
+		Enabled:   true,
+		Listen:    "0.0.0.0",
+		Port:      4222,
+		Advertise: "0.0.0.0",
+		NodeManager: types.NodeManager{
+			DisconnectTimeout: types.Duration(time.Second * 30),
+			AutoApprove:       true,
+		},
+		Scheduler: types.Scheduler{
+			Workers:              runtime.NumCPU(),
+			HousekeepingInterval: types.Duration(time.Second * 30),
+			HousekeepingTimeout:  types.Duration(time.Minute * 2),
+		},
+		Broker: types.EvaluationBroker{
+			VisibilityTimeout: types.Duration(time.Minute),
+			MaxRetries:        10,
+		},
+	},
+	Compute: types.Compute{
+		Enabled:       true,
+		Orchestrators: []string{"0.0.0.0:4222"},
+		Heartbeat: types.Heartbeat{
+			MessageInterval:  types.Duration(time.Second * 30),
+			ResourceInterval: types.Duration(time.Second * 30),
+			InfoInterval:     types.Duration(time.Second * 30),
+		},
+		Capacity: types.Capacity{
+			Allocated: types.ResourceScaler{
+				CPU:    "80%",
+				Memory: "80%",
+				Disk:   "80%",
+				GPU:    "100%",
+			},
+		},
+		Storages: types.Storage{
+			HTTP: types.HTTPStorage{
+				Enabled: true,
+			},
+		},
+		Engines: types.Engine{
+			Docker: types.Docker{
+				Enabled: true,
```

Comment:
Right, across the entire config I have adopted "positive logic" which I find is easier to reason about. It’s clearer to explicitly state what is enabled rather than what is disabled. This makes the configuration more declarative - if `Enabled` is `true`, the feature is on; if `Enabled` is `false`, the feature is off. It also aligns with Go’s default boolean values - when `Enabled` is absent the feature is off. And lastly, it avoid double negatives (i.e.., !Disabled), which are confusing and can be error-prone.

Reply:
I hear your point, and this makes our lives a little easier, but not for the users. Our goal is to have `bacalhau serve` to _just work_ with minimal user intervention. Therefore, auto-detecting the node's capabilities and installed engines is one of our goals. 

> It also aligns with Go’s default boolean values - when Enabled is absent the feature is off. And lastly, it avoid double negatives (i.e.., !Disabled), which are confusing and can be error-prone.
I don't think what you are doing here aligns with go or any language for that matter, as you are creating a boolean config but with a true default value. That is going to be confusing and will be error-prone. 

First, lets align on the engines being enabled by default or not, and if the answer is enabled, then the field name should be a negative, such as `Disabled` with default value being false so that empty value is the same as our expected default